Support the function COL_NAME()

roshan0708 opened the pull request:
### Description

Implement a function COL_NAME(table_id, column_id) which returns the Transact-SQL column name that corresponds to the provided column ID within the specified table.

For example:

Input-
CREATE TABLE test_table(c1 int);
GO
SELECT COL_NAME(OBJECT_ID('test_table'), 1);
GO

Output-
c1

Signed-off-by: Roshan Kanwar [rskanwar@amazon.com](mailto:rskanwar@amazon.com)

### Issues Resolved

BABEL-3172

### Test Scenarios Covered ###
* **Use case based -** Added use case based tests in JDBC framework


* **Boundary conditions -** Added edge cases where table_id and column_id both can be invalid or NULL


* **Arbitrary inputs -** Added cases for string/expression input


* **Negative test cases -** Added test cases where table_id and column_id will be negative


* **Minor version upgrade tests -**


* **Major version upgrade tests -**


* **Performance tests -**


* **Tooling impact -**


* **Client tests -**



### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

shalinilohia50 commented:
1. Please add this test to `upgrade/latest/schedule`
2. Please add dependency tests. Reference for dependency test: https://github.com/babelfish-for-postgresql/babelfish_extensions/pull/1701

hash-16 commented on test/JDBC/expected/BABEL-3172-vu-verify.out:
```diff
@@ -0,0 +1,58 @@
+SELECT * FROM sys.COL_NAME(NULL, 1);
+SELECT * FROM sys.COL_NAME(@table_id, -1);
+SELECT * FROM sys.COL_NAME(-1, 1);
+SELECT * FROM sys.COL_NAME(-1, -1);
```
Please add different test case which  include invalid inputs like strings,varbinary etc and also try some testcase where input is an expression and other such things.

shah-nirmit commented on contrib/babelfishpg_tsql/sql/sys_functions.sql:
```diff
@@ -3527,3 +3527,21 @@ CREATE OR REPLACE FUNCTION sys.fn_listextendedproperty
+            RETURN NULL;
+    END; 
+$$
+LANGUAGE plpgsql IMMUTABLE;
```
Suggestion:- Since we return NULL when either of arguments are NULL, can we define it as `STRICT`

shah-nirmit commented on test/JDBC/upgrade/latest/schedule:
```diff
@@ -189,6 +189,7 @@ bitwise_not-operator
 case_insensitive_collation
 cast_numeric_types_to_datetime
 cast_numeric_types_to_smalldatetime
+col_name
```
Nitpick:- would be better to have naming similar to object_id something like `BABEL_COL_NAME`

shalinilohia50 commented on test/JDBC/input/BABEL_COL_NAME-vu-prepare.sql:
```diff
@@ -0,0 +1,58 @@
+    id int,
+    names char
+)
+GO
```
Can you please create a table in another schema and write few tests around that?

hash-16 commented on test/JDBC/input/BABEL_COL_NAME-vu-prepare.sql:
```diff
@@ -0,0 +1,66 @@
+
+CREATE VIEW col_name_prepare_v1 AS (SELECT COL_NAME(CAST((SELECT OBJECT_ID('sys_column_name_vu_t_column_name')) AS INT), 1))
+GO
+
```
Can you please add testcase with hexadecimal, decimal , text values as input . And also try using NULL on both input parameter if not done already.

Commit:
Support the function COL_NAME()

Implement a function Implement a function COL_NAME(table_id, column_id)
which returns the Transact-SQL column name that corresponds to the provided column ID
within the specified table.

Task: BABEL-3172
Signed-off-by: Roshan Kanwar <rskanwar@amazon.com>
